WcaOnRails/app/views/registrations/edit.html.erb

```diff
@@ -13,10 +13,13 @@
     <% end %>
     <h4><%= @registration.name %></h4>
 
+    <% ineligible_events = @competition.competition_events.select{|e| !e.can_register?(@registration.user) }.map{|e| e.event } %>
```

Comment:
Same two nits as above. Why is this code duplicated anyways?

Reply:
That does make sense, but it shouldn't keep you from refactoring the common parts of the code into some helper method that can be invoked on both separate pages :)

Reply:
Ah, got it :) Added a method to Competition.